kork/kork-web/src/main/java/com/netflix/spinnaker/kork/client/ServiceClientProvider.java

```diff
@@ -34,7 +51,7 @@ public interface ServiceClientProvider {
    * @param <T> type of client , usually an interface with all the remote method definitions.
    * @return the retrofit interface implementation
    */
-  public <T> T getService(Class<T> type, ServiceEndpoint serviceEndpoint);
+  public <T> T getService(Class<T> type, ServiceEndpoint serviceEndpoint, RetrofitVersion version);
```

Comment:
We could significantly reduce the footprint of this change by having this interface have two signatures.  The implementation in `DefaultServiceClientProvider` can just have `RETROFIT2` be the default for the original signature.  Clients that have Retrofit1 annotations can use the second signature.

I think that a small footprint is desirable, since this change is just temporary, right?  We'd have to remove every `RETROFIT2` argument once Retrofit1 is fully removed.

Reply:
@jcavanagh - "defaulting to RETROFIT2" approach was considered but this PR is the beginning of a larger change of separating retrofit1 and retrofit2 beans. And instead of leaving doors open for mistakes it was decided to force the change for correctness even at expense of larger footprint. And this approach will be fool-proof when handling retrofit3 and later versions in future.

Reply:
Thoughts @jcavanagh ?  I wasn't planning to remove the enum argument once retrofit1 disappears, though I suppose we could.  I'd rather leave it in place (likely with a use of retrofit1 in test-only code) to pave the way for a future retrofit3.

Reply:
Yeah, given that we're planning to backport this to the release-2025.0.x branch, I like the idea of reducing breakage for folks.  Custom code that uses a retrofit1 interface with the no-enum version of this method that defaults to retrofit2 is already broken, but at least uses of retrofit2 interfaces with it will stay working.